stdlib/builtins.pyi

```diff
@@ -869,7 +869,11 @@ class dict(MutableMapping[_KT, _VT], Generic[_KT, _VT]):
         def __class_getitem__(cls, __item: Any) -> GenericAlias: ...
         def __or__(self, __value: Mapping[_T1, _T2]) -> dict[_KT | _T1, _VT | _T2]: ...
         def __ror__(self, __value: Mapping[_T1, _T2]) -> dict[_KT | _T1, _VT | _T2]: ...
-        def __ior__(self, __value: Mapping[_KT, _VT]) -> dict[_KT, _VT]: ...  # type: ignore[misc]
+        # dict.__ior__ should be kept roughly in line with MutableMapping.update()
+        @overload  # type: ignore[misc]
+        def __ior__(self: Self, __value: SupportsKeysAndGetItem[_KT, _VT]) -> Self: ...
+        @overload
+        def __ior__(self: Self, __value: Iterable[tuple[_KT, _VT]]) -> Self: ...
```

Comment:
Why is this not using a simple union?

Reply:
Because it has the same semantics as `MutableMapping.update()`, and there's a looong comment there explaining why that uses overloads rather than a union https://github.com/python/typeshed/blob/30580a373225cb747a03d6c7cecd487a7d5515b2/stdlib/typing.pyi#L474